framework/src/mesh/AnnularMesh.C

```diff
@@ -0,0 +1,230 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "AnnularMesh.h"
+
+#include "libmesh/face_quad4.h"
+#include "libmesh/face_tri3.h"
+
+template <>
+InputParameters
+validParams<AnnularMesh>()
+{
+  InputParameters params = validParams<MooseMesh>();
+  params.addRangeCheckedParam<unsigned int>(
+      "nr", 1, "nr>0", "Number of elements in the radial direction");
+  params.addRequiredRangeCheckedParam<unsigned int>(
+      "nt", "nt>1", "Number of elements in the angular direction");
+  params.addRequiredRangeCheckedParam<Real>(
+      "rmin",
+      "rmin>=0.0",
+      "Inner radius.  If rmin=0 then a disc mesh (with no central hole) will be created.");
+  params.addRequiredParam<Real>("rmax", "Outer radius");
+  params.addParam<Real>("tmin", 0.0, "Minimum angle, measured anticlockwise from x axis");
+  params.addParam<Real>("tmax",
+                        2 * M_PI,
+                        "Maximum angle, measured anticlockwise from x axis.  If "
+                        "tmin=0 and tmax=2Pi an annular mesh is created.  "
+                        "Otherwise, only a sector of an annulus is created");
+  params.addRangeCheckedParam<Real>(
+      "growth_r", 1.0, "growth_r>0.0", "The ratio of radial sizes of successive rings of elements");
+  params.addClassDescription("For rmin>0: creates an annular mesh of QUAD4 elements.  For rmin=0: "
+                             "creates a disc mesh of QUAD4 and TRI3 elements.  Boundary sidesets "
```

Comment:
Good! The combination of QUADs and TRIs will help with element quality but you have to be careful to assign them to different subdomains for Exodus support.

Reply:
Yep, done